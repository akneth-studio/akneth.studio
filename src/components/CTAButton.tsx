'use client'

import { Button } from "react-bootstrap";
import { useRouter } from "next/navigation";
import { ReactNode } from "react";

interface CTAButtonProps {
    text: string;
    icon?: ReactNode;
    variant?: 'primary' | 'outline-primary' | 'secondary' | 'outline-secondary' | string;
    size?: 'sm' | 'lg' | '';
    className?: string;
    to?: string;
    onClick?: () => void;
    type: 'button' | 'submit' | 'reset';
    tabIndex?: number;
    disabled?: boolean;
    "data-testid"?: string;
}

/**
 * Uniwersalny przycisk CTA do przekierowania na stronę kontaktową.
 */
export default function CTAButton({
    text,
    icon,
    to,
    onClick,
    variant = 'outline-primary',
    size = '',
    className = '',
    type = 'button',
    disabled = false,
    "data-testid": dataTestId = 'cta-button',
}: CTAButtonProps) {
    const router = useRouter();
    // Obsługa przekierowania, jeśli podano adres URL
    const handleClick = () => {
        // Jeśli jest podany onClick, użyj go
        if (onClick) {
            onClick();
        } else if (to) {
            router.push(to);
        }
    };
    return (
        <Button
            type={type}
            onClick={handleClick}
            className={`btn btn-${variant}${size ? ` btn-${size}` : ''} ${className}`}
            role="button"
            data-testid={dataTestId}
            tabIndex={0}
            aria-label={text}
            disabled={disabled}
        >
            {icon ? icon : null}
            {text}
        </Button>
    );
}