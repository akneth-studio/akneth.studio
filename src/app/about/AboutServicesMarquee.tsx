'use client';

import Image from 'next/image';
import CTAButton from '@/components/CTAButton';
import styles from '@/styles/AboutServicesMarquee.module.scss';

const iconsRow1 = [
    { src: '/icons/icons8-nextjs.svg', alt: 'Next.js' },
    { src: '/icons/react-logo.png', alt: 'React' },
    { src: '/icons/bootstrap-logo.svg', alt: 'Bootstrap' },
    { src: '/icons/icons8-sass-96.png', alt: 'SASS' },
];
const iconsRow2 = [
    { src: '/icons/icons8-office-365.svg', alt: 'Microsoft Office' },
    { src: '/icons/googleworkspace.svg', alt: 'Google Workspace' },
    { src: '/icons/icons8-javascript.svg', alt: 'JavaScript' },
    { src: '/icons/icons8-typescript.svg', alt: 'TypeScript' },
];

export default function AboutServicesMarquee() {
    return (
        <section className="about-services py-5 px-4">
            <div className="row align-items-center">
                {/* Tekst po lewej */}
                <div className="col-md-7 my-auto">
                    <h2 className="h4 mb-3">Zakres usług</h2>
                    <ul>
                        <li>Nowoczesne, dostępne strony internetowe (Next.js, React, TypeScript, Bootstrap, SCSS)</li>
                        <li>Szablony dokumentów w pakietach Office i Google Workspace (gotowe i na zamówienie)</li>
                        <li>Copywriting – teksty na strony, materiały informacyjne, treści SEO</li>
                        <li>Automatyzacje i integracje w Google Sheets</li>
                        <li>Szkolenia online z narzędzi biurowych i pracy zdalnej</li>
                        <li>Audyty dostępności (WCAG), wdrożenia RODO, własny Consent Manager</li>
                    </ul>
                    <CTAButton
                        type='button'
                        text="Zobacz pełną ofertę"
                        variant="outline-primary"
                        to="/services"
                    />
                </div>
                {/* Dwa rzędy marquee po prawej */}
                <div
                    className="col-md-5 d-flex flex-column justify-content-center align-items-center"
                    style={{ height: '320px' }}
                >
                    <div
                        style={{
                            display: 'flex',
                            flexDirection: 'column',
                            justifyContent: 'center',
                            alignItems: 'center',
                            gap: '32px',
                            width: '100%',
                            height: '100%',
                            overflow: 'hidden',
                            userSelect: 'none',
                        }}
                    >
                        {/* Rząd 1 */}
                        <div className={styles.scroll}>
                            {[...iconsRow1, ...iconsRow1].map((icon, idx) => (
                                <div key={idx} style={{ margin: '0 1.5rem', flex: '0 0 auto' }}>
                                    <Image
                                        src={icon.src}
                                        alt={icon.alt}
                                        width={100}
                                        height={100}
                                        style={{ height: '100px', width: 'auto', display: 'block' }}
                                        draggable={false}
                                    />
                                </div>
                            ))}
                        </div>
                        {/* Rząd 2 */}
                        <div className={styles.scrollReverse}>
                            {[...iconsRow2, ...iconsRow2].map((icon, idx) => (
                                <div key={idx} style={{ margin: '0 1.5rem', flex: '0 0 auto' }}>
                                    <Image
                                        src={icon.src}
                                        alt={icon.alt}
                                        height={100}
                                        width={100}
                                        style={{ height: '100px', width: 'auto', display: 'block' }}
                                        draggable={false}
                                    />
                                </div>
                            ))}
                        </div>
                    </div>
                </div>
            </div>
        </section>
    );
}
