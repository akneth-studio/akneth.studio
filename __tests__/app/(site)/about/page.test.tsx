/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render, screen } from '@testing-library/react';
import AboutPage from '@/app/(site)/about/page';
import AboutServicesMarquee from '@/components/AboutServicesMarquee';
import CTAButton from '@/components/CTAButton';

// Mock the components
jest.mock('@/components/AboutServicesMarquee', () => {
  return jest.fn(() => <div data-testid="about-services-marquee">About Services Marquee</div>);
});

jest.mock('@/components/CTAButton', () => {
  return jest.fn(() => <div data-testid="cta-button">CTA Button</div>);
});

// Mock environment variables
process.env.NEXT_PUBLIC_SITE_URL = require('../../../../jest.setup').NEXT_PUBLIC_SITE_URL || 'https://akneth.pl';

describe('AboutPage', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render all sections correctly', () => {
    render(<AboutPage />);
    
    // Check if all sections are rendered
    expect(screen.getByTestId('about-services-marquee')).toBeInTheDocument();
    expect(screen.getByTestId('cta-button')).toBeInTheDocument();
  });

  it('should render the correct layout structure', () => {
    render(<AboutPage />);
    
    // Check if the main container exists
    const container = screen.getByTestId('about-services-marquee').parentElement?.parentElement;
    expect(container).toBeInTheDocument();
    
    // Check if the header section exists
    const headerSection = screen.getByText('O mnie').closest('section');
    expect(headerSection).toBeInTheDocument();
    expect(headerSection).toHaveClass('about-hero');
    
    // Check if the values section exists
    const valuesSection = screen.getByText('Wartości i rozwój').closest('section');
    expect(valuesSection).toBeInTheDocument();
    expect(valuesSection).toHaveClass('about-values');
    
    // Check if the contact section exists
    const contactSection = screen.getByText('Porozmawiajmy!').closest('section');
    expect(contactSection).toBeInTheDocument();
    expect(contactSection).toHaveClass('about-contact');
  });

  it('should have correct metadata', () => {
    // Check if metadata is defined
    const { metadata } = require('@/app/(site)/about/page');
    expect(metadata).toBeDefined();
    expect(metadata.title).toBe('O mnie');
    expect(metadata.description).toBe('Dowiedz się więcej o AKNETH Studio i Katarzynie Pawłowskiej-Malesa. Oferujemy nowoczesne rozwiązania cyfrowe, które pomogą Ci w budowaniu profesjonalnego wizerunku online.');
    expect(metadata.keywords).toEqual([
      'AKNETH Studio',
      'Katarzyna Pawłowska-Malesa',
      'o mnie',
      'usługi cyfrowe',
      'wizerunek online',
      'automatyzacja',
      'React',
      'Next.js'
    ]);
    expect(metadata.alternates?.canonical).toBe('https://akneth.pl/about');
  });

  it('should match snapshot', () => {
    const { container } = render(<AboutPage />);
    expect(container).toMatchSnapshot();
  });
});
