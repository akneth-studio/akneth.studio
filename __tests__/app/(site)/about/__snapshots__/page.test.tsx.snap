// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`AboutPage should match snapshot 1`] = `
<div>
  <div
    class="d-flex row align-items-center justify-content-center py-4 px-3 about-header hero"
  >
    <div
      class="col-lg-4 mb-3 mb-lg-0 text-center"
    >
      <img
        alt="Katarzyna Pawłowska-Malesa, właścicielka AKNETH Studio"
        class="rounded-circle shadow mb-3 mb-lg-0"
        data-nimg="1"
        decoding="async"
        height="250"
        src="/_next/image?url=%2Fimg%2Fprofile.jpg&w=640&q=75"
        srcset="/_next/image?url=%2Fimg%2Fprofile.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg%2Fprofile.jpg&w=640&q=75 2x"
        style="color: transparent; object-fit: cover; object-position: center; vertical-align: middle;"
        width="250"
      />
    </div>
    <section
      class="about-hero align-content-center py-5 px-4 col-lg-8 my-auto"
    >
      <h1
        class="mb-3"
      >
        O mnie
      </h1>
      <p
        class="lead"
      >
        Nazywam się 
        <strong>
          Katarzyna Pawłowska-Malesa
        </strong>
         i prowadzę 
        <strong>
          AKNETH Studio
        </strong>
         - jednoosobową pracownię nowoczesnych rozwiązań cyfrowych. Pomagam firmom i klientom indywidualnym w budowaniu profesjonalnego wizerunku online oraz usprawnianiu codziennej pracy biurowej.
      </p>
    </section>
  </div>
  <div
    data-testid="about-services-marquee"
  >
    About Services Marquee
  </div>
  <div
    class="hr-about hero row mx-0 my-4 justify-content-center"
  >
    <section
      class="about-values py-5 px-4 col-md-6 mx-auto"
    >
      <h2
        class="h4 mb-3"
      >
        Wartości i rozwój
      </h2>
      <p>
        Stawiam na 
        <strong>
          transparentność
        </strong>
        , 
        <strong>
          dostępność
        </strong>
         i 
        <strong>
          partnerską współpracę
        </strong>
        . Rozwijam się w kierunku React, Next.js i automatyzacji, by stale poszerzać ofertę i dostarczać nowoczesne rozwiązania na miarę potrzeb rynku.
      </p>
    </section>
    <div
      class="col-md-1 d-none d-md-block"
    />
    <section
      class="about-contact py-5 px-4 col-md-5 mx-auto align-self-center text-center"
    >
      <h2
        class="h4 mb-3"
      >
        Porozmawiajmy!
      </h2>
      <p>
        Chcesz dowiedzieć się więcej lub rozpocząć współpracę?
         
      </p>
      <div
        data-testid="cta-button"
      >
        CTA Button
      </div>
    </section>
  </div>
</div>
`;
